### PR TITLE
update CEL filtering documentation

### PR DESCRIPTION
**Pull request by avinal**
# Changes

- update the filters list with currently supported filtering expressions
- add description and example for most used filters
- reformat markdown, fix tables and lint warnings

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/api/README.md [hunk 1]
```diff
@@ -175,37 +185,139 @@ status:
...
+
+#### Using Functions
+
+There are many functions that can be used in CEL expressions. Here is a
+list of functions that can be used in CEL expressions. The string in the function
+argument shows the expected type of the argument:
+
+| Functions                                             | Description                                                       | Example                                                        |
+| ----------------------------------------------------- | ----------------------------------------------------------------- | -------------------------------------------------------------- |
+| `startsWith('string')`                                | Checks if a string starts with a prefix                           | `name.startsWith("foo")`                                       |
```
Comment (khrm) [inline docs/api/README.md:265]: To be consistent, you should have `data.metadata.name.startswith("foo")`,

File: docs/api/README.md [hunk 2]
```diff
@@ -175,37 +185,139 @@ status:
...
+| `endsWith('string')`                                  | Checks if a string ends with a suffix                             | `name.endsWith("bar")`                                         |
```
Comment (khrm) [inline docs/api/README.md:266]: Same here.
```suggestion
| `endsWith('string')`                                  | Checks if a string ends with a suffix                             | `data.metadata.name.endsWith("bar")`                                         |
```

File: docs/api/README.md (hunk 1 above)
Comment (khrm) [inline docs/api/README.md:265]:
```suggestion
| `startsWith('string')`                                | Checks if a string starts with a prefix                           | `data.metadata.name.startswith.startsWith("foo")`                                       |
```

Comment (avinal) [reply docs/api/README.md:265]: Umm I wanted to show the function on `name` field of record, but I will change it.

**Comment by avinal** [comment]
cc: @khrm @enarha @sayan-biswas

**Review by khrm — COMMENTED** [review]
/approve

**Review by adambkaplan — APPROVED** [review]
/lgtm

This is fantastic - great work @avinal! :tada:

**Pull request merged**